File: MongoDB/JavaODMs/odmstest/src/main/java/cz/cuni/mff/mongodb_java/springdata/models/Employee.java
package cz.cuni.mff.mongodb_java.springdata.models;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.aggregation.ArithmeticOperators;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;

@Document(collection = "employees")
public class Employee {
    @Id
    private ObjectId id;

    private String name;
    private int age;

    @DBRef
    private Employee manager;

    private List<String> emails;
    private Double salary;

    private Address address;

    public Employee(String name, int age, Employee manager, List<String> emails, Double salary, Address address) {
        this.name = name;
        this.age = age;
        this.manager = manager;
        this.emails = emails;
        this.salary = salary;
        this.address = address;
    }

    public String getName() { return name; }

}

File: MongoDB/CSharpODMs/MongoDBEntities/MongoDBEntities/Models/Employee.cs
using MongoDB.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace MongoDBEntities.Models
{
    internal class Employee : Entity
    {
        public string Name { get; set; }
        public int Age { get; set; }
        public One<Employee> Manager { get; set; }
        public List<string> Emails { get; set; }
        public double Salary { get; set; }
        public Address Address { get; set; }

        public string GetName => Name;
    }
}
